Sources/RInAppMessaging/UserInfoProvider.swift

```diff
@@ -66,8 +66,14 @@ func == (lhs: UserInfoProvider?, rhs: UserInfoProvider?) -> Bool {
 
 @inlinable
 func != (lhs: UserInfoProvider?, rhs: UserInfoProvider?) -> Bool {
-    let equalsResult = lhs == rhs
-    return !equalsResult
+    switch (lhs, rhs) {
```

Comment:
Was this change necessary?

Reply:
Yes, otherwise the code on 87 - 91 will not get called:
```
@inlinable
func != (lhs: UserInfoProvider, rhs: UserInfoProvider) -> Bool {
    let equalsResult = lhs == rhs
    return !equalsResult
}
```

Reply:
Hmm. But it would if you used non optional `UserInfoProviderMock` objects in your test, wouldn't it?

Reply:
but maybe it's better to keep it your way after all